src/main/java/nl/ordina/jobcrawler/service/SkillService.java

```diff
@@ -97,7 +104,7 @@ public void removeSkillsFromVacancy(Set<Skill> skills, Vacancy vacancy) {
 
 
     //******** Updating ********//
-    // Takes a new set of skills to replace old ones from a vacancy
+    // Takes a new set of skills to replace old ones from a vacancy in the database
```

Comment:
This feels like it should go into the VacancyService, because you're not updating any skills, rather re-linking them to a specific vacancy. I would move this to the VacancyService instead. There it should be used as:

`myVacancy.setSkills(Set<Skill> replaceSkills)`
or 
`
myVacancy.addSkill(Skill extraSkill)
myVacancy.addSkills(Set<Skill> extraSkills)
`

Save them with:
`vacancyService.save(myVacancy)`

Reply:
adapted